app/bundles/EmailBundle/Tests/Model/SendEmailToContactTest.php

```diff
@@ -725,4 +725,102 @@ function () use ($emailMock) {
         $errorMessages = $model->getErrors();
         $this->assertCount(1, $errorMessages);
     }
+
+    public function testSetSampleMailerByTemporaryMailer()
+    {
+        // Use our test token transport limiting to 1 recipient per queue
+        $transport = new BatchTransport(true, 1);
+        $mailer    = new \Swift_Mailer($transport);
+
+        // Mock factory to ensure that queue mode is handled until MailHelper is refactored completely away from MauticFactory
+        $factoryMock = $this->getMockBuilder(MauticFactory::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $factoryMock->expects($this->at(3))->method('getParameter')->with('mailer_spool_type')->willReturn('memory');
+        $factoryMock->expects($this->at(1))->method('getParameter')->willReturn('');
+
+        $factoryMock->method('getLogger')
+            ->willReturn(
+                new NullLogger()
+            );
+
+        $mockEm = $this->createMock(EntityManager::class);
+
+        $factoryMock->method('getEntityManager')
+            ->willReturn($mockEm);
+
+        $mockDispatcher = $this->getMockBuilder(EventDispatcher::class)
+            ->getMock();
+        $mockDispatcher->method('dispatch')
+            ->willReturnCallback(
+                function ($eventName, EmailSendEvent $event) {
+                    $lead = $event->getLead();
+
+                    $tokens = [];
+                    foreach ($lead as $field => $value) {
+                        $tokens["{contactfield=$field}"] = $value;
+                    }
+                    $tokens['{hash}'] = $event->getIdHash();
+
+                    $event->addTokens($tokens);
+                }
+            );
+        $factoryMock->method('getDispatcher')
+            ->willReturn($mockDispatcher);
+        $routerMock = $this->getMockBuilder(Router::class)
+            ->disableOriginalConstructor()
+            ->getMock();
```

Comment:
These 3 rows can be simplified into `$factoryMock = $this->createMock(Router::class);`